File: Java_Collections_Benchmark/src/com/company/ResultsManager.java
package com.company;

import java.io.*;
import java.util.List;

public class ResultsManager {

    private final String collectionInfo = "Collection name: %1$s,number of operations: %2$s,unit: %3$s";
    private final String columns = "Add time,Find time,Remove time";
    private final String rowPattern = "%1$s,%2$s,%3$s";
    private final String measureUnit = "ms";
    public String CollectionName = "None";
    public int NumberOfOperations = 0;


    public void SaveToCsv(List<BenchmarkResults> results, String path) throws IOException {
        StringBuilder csv = new StringBuilder();
        String firstRow = String.format
                (collectionInfo, CollectionName, NumberOfOperations, measureUnit);

        String newLine = System.lineSeparator();
        csv.append(firstRow).append(newLine);
        csv.append(columns).append(newLine);

        for (int i = 0; i < results.size(); i++)
        {
            String row = String.format
                    (rowPattern,
                            results.get(i).AddTime, results.get(i).FindTime, results.get(i).RemoveTime);

            csv.append(row).append(newLine);
        }

        BufferedWriter writer = new BufferedWriter(new FileWriter(path));
        writer.write(csv.toString());

        writer.close();

    }
}

File: Python_Collections_benchmark/ResultsManager.py


class ResultsManager():
    _collection_info = 'Collection name: {},number of operations: {},unit: {}'
    _columns = 'Add time,Find time,Remove time'
    _row_pattern = '{},{},{}'
    _measure_unit = "ms"

    collection_name = "None"
    number_of_operations = 0

    def save_to_csv(self,results :list, path:str):
        csv = list()
        first_row = self._collection_info.format\
            (self._collection_info, self.collection_name, self.number_of_operations, self._measure_unit)

        csv.append(first_row)
        csv.append(self._columns)

        for item in results:
            row = self._row_pattern.format(str(item.add_time), str(item.find_time), str(item.remove_time))
            csv.append(row)

        print('saving to file:' + self.collection_name)
        output = '\n'.join(csv)

        with open(path,'w') as file:
            file.write(output)

